### PR TITLE
Remove Math::Clamp in favour of std::clamp

### PR DESCRIPTION
**Pull request by LinusU**
As discussed in #7868, now that we are targeting C++17 we can use `std::clamp` instead of our own version. I have switched the order of the arguments in order to match `std::clamp(value, min, max)` instead of `Math::Clamp(min, value, max)`.

### CODE REVIEW

**Comment by LinusU** [comment]
Turns out that there was only one usage of `Math.hpp` left, that used `Math::Sign`. In that code path, the value `0` was already checked for so I replaced it with a simple ternary expression.

File: src/openrct2-ui/audio/AudioChannel.cpp [hunk 1]
```diff
@@ -15,7 +15,6 @@
 #include <openrct2/audio/AudioChannel.h>
 #include <openrct2/audio/AudioSource.h>
 #include <openrct2/common.h>
-#include <openrct2/core/Math.hpp>
```
Comment (Broxzier) [inline src/openrct2-ui/audio/AudioChannel.cpp]: Many of the files don't include <algorithm> directly now, which is where `std::clamp` is defined.

File: src/openrct2/scenario/ScenarioRepository.cpp [hunk 2]
```diff
@@ -48,7 +47,7 @@ static int32_t ScenarioCategoryCompare(int32_t categoryA, int32_t categoryB)
         return -1;
     if (categoryB == SCENARIO_CATEGORY_BUILD_YOUR_OWN)
         return 1;
-    return Math::Sign(categoryA - categoryB);
+    return (categoryA < categoryB) ? -1 : 1;
```
Comment (AaronVanGeffen) [inline src/openrct2/scenario/ScenarioRepository.cpp]: Math::Sign would return 0 when the input was 0.

**Comment by AaronVanGeffen** [comment]
Thanks for tackling this. Your changes look good to me, barring a minor change as a result of removing Math::Sign.

File: src/openrct2/scenario/ScenarioRepository.cpp (hunk 2 above)
Comment (Broxzier) [reply src/openrct2/scenario/ScenarioRepository.cpp]: OP mentioned that in the comment, it's being checked at the top of this function. I'm not really in favour of replacing the function with a ternary operator though. I'd rather see an easier-to-read branch instead, like the other statements above.

File: src/openrct2-ui/audio/AudioChannel.cpp (hunk 1 above)
Comment (LinusU) [reply src/openrct2-ui/audio/AudioChannel.cpp]: Oh 🤦‍♂️, on it!

File: src/openrct2/scenario/ScenarioRepository.cpp (hunk 2 above)
Comment (LinusU) [reply src/openrct2/scenario/ScenarioRepository.cpp]: That is handled at the first line of this function

**Comment by LinusU** [comment]
Found two places where the argument order was flipped in the first place, corrected them in 5a48519.

**Comment by Broxzier** [comment]
`Sign` is a function that is not part of the standard. Even while it's not being used in the codebase anymore, I think it can't hurt to keep it.

**Comment by AaronVanGeffen** [comment]
> Merge branch 'develop' into clamp

Please rebase your branch instead. Thanks!

**Comment by LinusU** [comment]
> Please rebase your branch instead.

Sure thing 👍 it's just so convenient to fix merge conflicts using the GitHub UI 😆 

> Even while it's not being used in the codebase anymore, I think it can't hurt to keep it.

Hmm, to me it seems very strange to keep around code that is no longer needed. I did a search thru the git history and this seems to be the only place that this function has ever been called, seems like the odds of us needing this function in the future isn't too high...

Also, it's very easy to find an (even better) implementation with a quick google search: https://stackoverflow.com/a/4609795/148072

just my two cents...

**Comment by LinusU** [comment]
Rebased and cleaned up the commits as well 🐎

**Comment by LinusU** [comment]
Fixed the formatting errors and rebased on master again 🐎

**Comment by AaronVanGeffen** [comment]
Our CI reports a few [formatting errors](https://travis-ci.org/OpenRCT2/OpenRCT2/jobs/415112542) (clang-format). Could you address them too, please?

**Comment by LinusU** [comment]
Seems like the build failed with a network error:

```text
    for file: [/home/travis/build/OpenRCT2/OpenRCT2/build/OpenRCT2/share/openrct2/title/title-sequences.zip]
      expected hash: [1136ef92bfb05cd1cba9831ba6dc4a653d87a246]
        actual hash: [da39a3ee5e6b4b0d3255bfef95601890afd80709]
             status: [7;"Couldn't connect to server"]
```

@AaronVanGeffen would you mind restarting that build? 💌

**Comment by Broxzier** [comment]
So now `Math.hpp` is completely gone, I still think `Math::Sign` would have been nice to keep...

**Comment by AaronVanGeffen** [comment]
I agree with @LinusU that the implementation wasn't that general to begin with, and seeing as it was only used in one place in the code… I'm not mourning it, to be honest.

**Comment by IntelOrca** [comment]
`Math::Sign` can be reinstated when it is needed again, otherwise we have have a file which is never included and contains one unused function which is very simple.

**Pull request merged**